File: 1회차/programmers_algorithm/5.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, vector<vector<int>> dungeons) {
    int answer = -1;
    int n = dungeons.size();
    int p[n];
    for (int i = 0; i < n; i++) p[i]=i;
    do {
        int cnt = 0;
        int tmpk = k;
        for (int i = 0; i < n; i++) {
            if (tmpk >= dungeons[p[i]][0]) {
                tmpk-=dungeons[p[i]][1];
                cnt++;
            } else break;
        }
        answer=max(answer,cnt);
    } while (next_permutation(p,p+n));
    return answer;
}

File: 1회차/programmers_algorithm/5.py
from itertools import permutations
def solution(k, dungeons):
    answer = -1
    n = len(dungeons)
    a=[]
    for i in range(n):
        a.append(i)
    p = list(permutations(a,n))
    for i in range(len(p)):
        cnt=0
        tmpk=k
        for j in range(n):
            if tmpk >= dungeons[p[i][j]][0]:
                tmpk-=dungeons[p[i][j]][1]
                cnt+=1
            else:
                break
        answer=max(answer,cnt)
    return answer